namespace/data.go

```diff
@@ -6,6 +6,7 @@ package namespace
 // Go's type system does not allow enforcing the structure we want:
 // [namespaceID, rawData ...], especially as this type does not expect any
 // particular size for the namespace.
+// TODO [Me] Shouldn't we specify that the first 8 bytes represent the namespace.ID
 type PrefixedData []byte
```

Comment:
Yeah, the issue with this is that the library wants to support varying namespace sizes (not only 8 bytes necessarily). 

We could instead:
 - just not use a type alias for byte and simply accept that this is even easier to misuse (would have the nice side effect that the user would not need to cast between bytes and PrefixedData at all)
 - use a dedicated type; e.g. one of the two options here: https://github.com/celestiaorg/nmt/issues/71